plugins/modules/foreman_search_facts.py

```diff
@@ -105,7 +105,7 @@ def main():
         supports_check_mode=True,
     )
 
-    module_params = module.parse_params()
+    module_params = module.clean_params()
```

Comment:
why that rename?

Reply:
I do not remember. Maybe an attempt to keep the old interface intact for now.

Reply:
I guess, it's not needed.

Reply:
Now i remember. `parse_params` returns the cleaned parameters _and_ the state for some module classes. So i needed a new function that does not get overwritten by the subclasses.